### DISCUSSION TITLE
rotate annotations together with tensor

### DISCUSSION THREAD

**Discussion by lfgmorales** [mtex-toolbox/mtex#722] [Ask Anything] [answered]
Hi all

If I have 
```
%% Define tensor reference frame

cs_tensor_ort = crystalSymmetry('12/m1',[8.561, 12.996, 7.192],...
    [90, 116.01, 90]*degree, 'mineral','orthoclase','Y||b','Z||c') 

%% Define thermal conductivity tensor coefficients (in W/m/K unit)

M = [[1.45 0.00 0.19];[0.00 2.11 0.00];[0.19 0.00 1.79]]

%%  Define the thermal conductivity tensor

k = tensor (M, 'name', 'thermal_conductivity','rank',2, 'unit', ...
    'W_1/m_1/K',cs_tensor_ort)

%% Plot thermal conductivity tensor

plot (k, 'complete','lower','contourf','minmax')
mtexColorbar

% annotate with single crystal pole to planes

hold on
text(Miller({1,0,0},{0,1,0},{0,0,1},cs_tensor_ort), {'(100)','(010)',...
    '(001)'},'backgroundColor','w','fontweight','bold','fontsize',18)
hold off
```
I have this plot

<img width="563" alt="Screenshot 2021-01-24 at 18 51 56" src="https://user-images.githubusercontent.com/46781495/105639798-7c568c80-5e7a-11eb-891e-fa56c14d35e4.png">
 
Now if I rotate this tensor and replot, the crystal reference frame does not rotate together
```
%% Let's rotate the tensor and plot it again

% define a rotation

g = orientation.byEuler(10*degree,20*degree,30*degree,cs_tensor_ort)

% rotate tensor

k_rot = rotate (k,g)

% plot rotated tensor

plot (k_rot, 'complete','lower','contourf')
mtexColorbar

% annotate with single crystal pole to planes

hold on
text(Miller({1,0,0},{0,1,0},{0,0,1},cs_tensor_ort), {'(100)','(010)',...
    '(001)'},'backgroundColor','w','fontweight','bold','fontsize',18)
hold off

% don't forget to save your figures

saveFigure ('thermal conductivity tensor orthoclase rotated.png')`
```
<img width="624" alt="Screenshot 2021-01-24 at 19 30 47" src="https://user-images.githubusercontent.com/46781495/105639851-ba53b080-5e7a-11eb-87ac-7322753f3a19.png">

Is it possible to rotate the crystal reference frame together with the tensor property?

cheers

Luiz

**Comment by kilir** [comment, marked as answer]
Hi Luiz,
I guess if you'd like to rotate the tensor and it should stay a matter tensor, it should be a rotation, not an orientation (check the symmetry'mineral' of the tensor after rotating):
 
```
g = rotation.byEuler(10*degree,20*degree,30*degree)
```
and the text  could end up with the same rotation to the corresponding postions.

```
text(g* Miller({1,0,0},{0,1,0},{0,0,1},cs_tensor_ort), {'(100)','(010)',...
    '(001)'},'backgroundColor','w','fontweight','bold','fontsize',18)
```

Does that do what you want?

Cheers,
Rüdiger

**Reply by lfgmorales** [reply]
Dear Sir

Yes, I realized that rotation was the correct option, removing cs (but orientation and rotation generate the same matrix), but I was missing g *Miller, thanks a lot Sir 

cheers


L
